File: codegen/codegen-erlang/src/test/resources/ir/aws_json_encode_get_user_request.expected.erl
%% @doc Encode AWS JSON request for smithy.beam.test.awsjson11#GetUser.
-spec encode_get_user_request(get_user_input()) -> #http_request{}.
encode_get_user_request(Input = #get_user_input{user_name = UserName}) ->
    BodyMap = maps:filter(
        fun(_, V) ->
            V =/= undefined
        end,
        #{<<"userName">> => UserName}
    ),
    Body = jsone:encode(BodyMap),
    #http_request{
        method = <<"POST">>,
        path = <<"/">>,
        query = #{},
        headers = [
            {<<"Content-Type">>, <<"application/x-amz-json-1.1">>},
            {<<"X-Amz-Target">>, <<"Json11Service.GetUser">>}
        ],
        body = Body
    }.

File: codegen/codegen-elixir/src/test/resources/ir/aws_json_encode_get_user_request.expected.ex
@doc "Encode AWS JSON request for smithy.beam.test.awsjson11#GetUser."
@spec encode_get_user_request(Json11serviceTypes.GetUserInput.t()) :: %RuntimeTypes.HttpRequest{}
def encode_get_user_request(input = %Types.GetUserInput{}) do
  body_map =
    %{"userName" => input.user_name}
    |> Enum.reject(fn {_, v} -> Kernel.is_nil(v) end)
    |> Map.new()
  body = Jason.encode!(body_map)
  %RuntimeTypes.HttpRequest{
    method: "POST",
    path: "/",
    query: %{},
    headers: [{"Content-Type", "application/x-amz-json-1.1"}, {"X-Amz-Target", "Json11Service.GetUser"}],
    body: body
  }
end